writer.go

```diff
@@ -96,3 +102,45 @@ func MultiLevelWriter(writers ...io.Writer) LevelWriter {
 	}
 	return multiLevelWriter{lwriters}
 }
+
+// TestWriter is a writer that writes to testing.TB.
+type TestWriter struct {
+	TB testing.TB
+
+	// Frame skips caller frames to capture the original file and line numbers.
+	Frame int
+}
+
+// NewTestWriter creates a writer that logs to the testing.TB.
+func NewTestWriter(t testing.TB) TestWriter {
+	return TestWriter{TB: t, Frame: 1}
+}
+
+// Write to testing.TB.
+func (t TestWriter) Write(p []byte) (n int, err error) {
+	n = len(p)
+
+	// Strip trailing newline because t.Log always adds one.
+	p = bytes.TrimRight(p, "\n")
+
+	// Try to correct the log file and line number to the caller.
+	if t.Frame > 0 {
```

Comment:
Wouldn't `t.TB.Helper()` do that for you?

Reply:
Helper moves the problem up one, but there’s too many function calls that don’t know about the ‘t.Helper()’. For example, the ConsoleLogger with helper will set the log prefix to: ‘buffer.go:293: ‘ as it calls bytes buffer WriteTo method on the writer.

Reply:
Changed to calling `t.Helper` to have a better zero frame default.